src/MathOptInterface.jl

```diff
@@ -164,27 +164,31 @@ function copy_to(dest, src; kwargs...)
 end
 
 """
-    copy_to_and_optimize!(
+    optimize_model!(
         dest::AbstractOptimizer,
         src::ModelLike,
     )::IndexMap
 
-A single call equivalent to calling [`copy_to`](@ref) followed by
-[`optimize!`](@ref).  Like [`copy_to`](@ref), it returns an [`IndexMap`].
-
-Keyword arguments are passed to [`copy_to`](@ref).
+A single call similar to calling [`copy_to`](@ref) followed by
+[`optimize!`](@ref). Return an [`IndexMap`](@ref) and a `Bool` `copied`.
+If `copied` is `true`, `src` was copied to `dest` so [`optimize!`](@ref)
+can be called again on `dest`, e.g., after changing the starting values
+or optimizer attributes. On the other hand, if `copied` is `false`, the
+model was not copied to `dest` so calling [`optimize!`](@ref) is not allowed.
+However, attributes for which [`is_set_by_optimize`](@ref) is true can be
+queried from `dest`.
```

Comment:
I think we need a 
```Julia
!!! note
    The main purpose of this function is for use in [`Utilities.CachingOptimizer`](@ref).
```
Or something similar? We should say _why_ this function exists, because it's a little unusual.